File: AlgoExpert/Arrays/05-Non-Constructible-Change/Java/Solution.java
import java.util.*;

class Solution {

  public int nonConstructibleChange(int[] coins) {
    // Write your code here.		
    Arrays.sort(coins); // nLog(n) runtime 
    
    if (coins == null || coins.length == 0 || coins[0] != 1) {
        return 1;
    }
    
    int change = 0;
    
    for (int coin : coins) {
        if (coin > (change + 1)) {
            return change + 1;
        }	
        change += coin;
    }
    
    return change + 1;	
  }
}

File: AlgoExpert/Arrays/05-Non-Constructible-Change/Python/solution.py
def nonConstructibleChange(coins):
    # Write your code here.
    coins.sort()

    if coins == None or len(coins) == 0 or coins[0] != 1:
        return 1

    change = 0

    for coin in coins:
        if coin > change + 1:
            return change + 1
        change += coin
        
    return change + 1
